apex/contrib/test/group_norm/test_group_norm.py

```diff
@@ -0,0 +1,153 @@
+# FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import torch
+import unittest
+
+SKIP_TEST = None
+try:
+    from apex.contrib.group_norm.group_norm import cuda_group_norm_nhwc_one_pass
+    from apex.contrib.group_norm.group_norm import cuda_group_norm_nhwc_two_pass
+except ImportError as e:
+    SKIP_TEST = e
+
+
+# pytorch group norm requires same input type
+def torch_group_norm(x, g, w, b, eps, act=""):
+    xdtype, wdtype = x.dtype, w.dtype
+    if xdtype != wdtype:
+        x = x.to(dtype=wdtype)
+    y = torch.nn.functional.group_norm(x, g, w, b, eps)
+    if act in ["silu", "swish"]:
+        y = torch.nn.functional.silu(y)
+    if xdtype != wdtype and y.dtype != xdtype:
+        y = y.to(dtype=xdtype)
+    return y
+
+
+@unittest.skipIf(SKIP_TEST, f"{SKIP_TEST}")
+class GroupNormTest(unittest.TestCase):
+
+    def setUp(self, seed=0):
+        super().setUp()
+        torch.manual_seed(seed)
+
+    def verify_group_norm(self,
+                          tst_func,
+                          N=32,
+                          C=128,
+                          H=256,
+                          W=256,
+                          G=32,
+                          ref_func=torch_group_norm,
+                          xdtype=torch.float16,
+                          wdtype=torch.float32,
+                          eps=1e-5,
+                          memory_format=torch.channels_last,
+                          device='cuda',
+                          act=""):
+        # create data
+        x_shape = (N, C, H, W)
+        w_shape = (C, )
+        weight = torch.rand(w_shape,
+                            dtype=wdtype,
+                            device='cuda',
+                            requires_grad=True)
+        bias = torch.rand(w_shape,
+                          dtype=wdtype,
+                          device='cuda',
+                          requires_grad=True)
+        x = -2.3 + 0.5 * torch.randn(x_shape, dtype=xdtype, device='cuda')
+        x = x.to(memory_format=memory_format)
+        dy = .1 * torch.randn_like(x)
+        x.requires_grad_(True)
+
+        # forward pass
+        y_ref = ref_func(x, G, weight, bias, eps, act)
+        y_tst = tst_func(x, G, weight, bias, eps, act)
+        # backward pass
+        y_ref.backward(dy, retain_graph=True)
+        dx_ref, dw_ref, db_ref = [_.grad.clone() for _ in [x, weight, bias]]
```

Comment:
this is my taste, but I'd not use `_` if it's really used
```suggestion
        dx_ref, dw_ref, db_ref = [t.grad.clone() for t in [x, weight, bias]]
```